### ISSUE TITLE
Update markup and signature exit model with new styling 

### ISSUE THREAD

**Issue by alyblenkin** [getodk/collect#6117]
If you go back without saving markup and signature changes the dialog isn't using our new styling.

![Image](https://github.com/getodk/collect/assets/7871340/f474f290-99fe-497a-8718-81c7c5b109ae)

[Fimga
](https://www.figma.com/file/KFi9hIQdrRgo0rLqIC7Sg8/ODK-user-flows?type=design&node-id=4952%3A38299&mode=design&t=ErYDN945bNvJMVwh-1)

**Comment by alyblenkin** [comment]
Just came across this one again! It will be nice to make all the dialogs consistent.

**Comment by grzesiek2010** [comment]
@seadowg the pull request is only blocked by tests, so le's maybe create a separate issue for adding them. What do you think?
